src-java/kilda-pce/src/main/java/org/openkilda/pce/impl/InMemoryPathComputer.java

```diff
@@ -217,20 +219,33 @@ public List<Path> getNPaths(SwitchId srcSwitchId, SwitchId dstSwitchId, int coun
                 .collect(Collectors.toList());
     }
 
-    private WeightFunction getWeightFunctionByStrategy(PathComputationStrategy strategy) {
+    private WeightFunction getWeightFunctionByStrategy(PathComputationStrategy strategy, Flow flow) {
         switch (strategy) {
             case COST:
                 return this::weightByCost;
             case LATENCY:
             case MAX_LATENCY:
-                return this::weightByLatency;
+                return getMaxLatencyWeightFunction(flow);
             case COST_AND_AVAILABLE_BANDWIDTH:
                 return this::weightByCostAndAvailableBandwidth;
             default:
                 throw new UnsupportedOperationException(String.format("Unsupported strategy type %s", strategy));
         }
     }
 
+    /**
+     * Check if this function will be used for finding protected
+     * path and return appropriate WeightFunction.
+     */
+    private WeightFunction getMaxLatencyWeightFunction(Flow flow) {
+        if (flow.isAllocateProtectedPath()
+                && flow.getForwardPath() != null && flow.getReversePath() != null) {
```

Comment:
I still want to have more obvious and durable way to understand that we need protected path.